tests/gen_cert_test.py

```diff
@@ -181,3 +216,70 @@ def test_cert_date(self, force_date, locale, timezone, expected):
         """
         cert_date = get_cert_date(force_date, locale, timezone)
         assert_equal(expected, cert_date)
+
+
+class CertificateSubtemplateTests(unittest.TestCase):
+    """
+    Tests the subtemplates feature for certs
+
+    Subtemplates are currently written to key on designations only
+
+    These tests have a specific order but do not need to be run in order.
+    They were split and ordered to more easily identify errors while testing.
+    """
+    subtemplate_courses = [
+        'Test/Subtemplates/ByDesignations',
+    ]
+
+    def test_subtemplate_courses_exist(self):
+        """
+        Check that the subtemplate courses exist in settings.CERT_DATA
+        """
+        found_sub_course_ids = []
+        for course_id, course_data in settings.CERT_DATA.iteritems():
+            if course_data.get('subtemplates'):
+                found_sub_course_ids.append(course_id)
+        assert_equal(
+            set(self.subtemplate_courses),
+            set(found_sub_course_ids)
+        )
+
+    def test_subtemplate_courses_have_proper_data(self):
+        """
+        Check that subtemplates have proper key-value pairs in 'subtemplates'
+
+        Keys should be designations
+        Values should be course ids in settings.CERT_DATA
+        """
+        for course_id in self.subtemplate_courses:
+            cert_data = settings.CERT_DATA[course_id]
+            subtemplates = cert_data['subtemplates']
+            assert_true(isinstance(subtemplates, dict))
+            for designation, sub_course_id in subtemplates.iteritems():
+                try:
+                    assert_true(settings.CERT_DATA.get(sub_course_id))
+                except AssertionError as e:
+                    e.args += ('Course id not found: ' + sub_course_id,)
+                    raise
+
+    def test_subtemplates(self):
+        """
+        Generates certficates with subtemplates
+        """
+        for course_id in self.subtemplate_courses:
+            cert_data = settings.CERT_DATA[course_id]
+            subtemplates = cert_data['subtemplates']
+            for designation, course_id in subtemplates.iteritems():
+                tmpdir = tempfile.mkdtemp()
+                cert = CertificateGen(course_id, designation=designation)
+                for name in NAMES:
+                    (download_uuid,
+                        verify_uuid, download_url) = cert.create_and_upload(
+                        name,
+                        upload=False,
+                        copy_to_webroot=True,
+                        cert_web_root=tmpdir,
+                        cleanup=True,
+                    )
+                    if os.path.exists(tmpdir):
+                        shutil.rmtree(tmpdir)
```

Comment:
I didn't write this code block.  It is the standard way we accomplish this task as seen in the test_cert_gen.  

Fixing how we create and delete test certs is not in the scope of this feature.  We are already making a trello card to fix cert tests so we can put it in there.